solana/write-account/src/lib.rs

```diff
@@ -53,39 +82,86 @@ solana_program::entrypoint!(process_instruction);
 /// - `offset` → zero (write from the start of the first account),
 /// - `algo` → zero (null compression).
 pub fn process_instruction(
-    _program_id: &Pubkey,
+    program_id: &Pubkey,
     accounts: &[AccountInfo],
     mut instruction: &[u8],
 ) -> Result {
     match instruction.unshift().ok_or(ProgramError::InvalidInstructionData)? {
-        0 => handle_write(accounts, instruction),
-        1 => handle_copy(accounts, instruction),
+        0 => handle_create(program_id, accounts, instruction),
+        1 => handle_write(program_id, accounts, instruction),
+        2 => handle_copy(accounts, instruction),
         _ => Err(ProgramError::InvalidInstructionData),
     }
 }
 
+/// Handles Create Operation. See [`process_instruction`].
+fn handle_create(
+    program_id: &Pubkey,
+    accounts: &[AccountInfo],
+    mut data: &[u8],
+) -> Result {
+    let accounts_iter = &mut accounts.iter();
+    let write_account = next_account_info(accounts_iter)?;
+    let payer = next_account_info(accounts_iter)?;
+    let system_program = next_account_info(accounts_iter)?;
 
-/// Handles a Write operation.  See [`process_instruction`].
-fn handle_write(accounts: &[AccountInfo], mut data: &[u8]) -> Result {
-    let account = match accounts {
-        [account, ..] if account.is_writable => Ok(account),
-        [_, ..] => Err(ProgramError::InvalidAccountData),
-        _ => Err(ProgramError::NotEnoughAccountKeys),
-    }?;
-
-    let offset = data
+    let account_span = data
         .unshift_n::<4>()
         .ok_or(ProgramError::InvalidInstructionData)
         .and_then(usize_from_bytes)?;
+    let lamports_required = Rent::get()?.minimum_balance(account_span);
+    let (_pubkey, bump) =
+        Pubkey::find_program_address(&[payer.key.as_ref(), data], program_id);
+
+    invoke_signed(
+        &system_instruction::create_account(
+            payer.key,
+            write_account.key,
+            lamports_required,
+            account_span as u64,
+            program_id,
+        ),
+        &[payer.clone(), write_account.clone(), system_program.clone()],
+        &[&[payer.key.as_ref(), data, &[bump]]],
+    )?;
```

Comment:
Shouldn’t this be [create_account_with_seed](https://docs.rs/solana-program/latest/solana_program/system_instruction/fn.create_account_with_seed.html) instead?

Reply:
`create_account_with_seed` seems to be different than creating PDA. But i was thinking, we just need an account with seed and we can create it on client side as well. So maybe we can remove this instruction as well if we use those.